packages/ui/src/utils/fast-deep-merge.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Merges 2 objects without creating new object references
+ * The merged props will appear on the `target` object
+ * If `target` already has a value for a given key it will not be overwritten
+ */
+export const fastDeepMergeAndReplace = (
+  source: Record<any, any> | undefined | null,
+  target: Record<any, any> | undefined | null,
+) => {
+  if (!source || !target) {
+    return;
+  }
+
+  for (const key in source) {
+    if (source[key] !== null && typeof source[key] === `object`) {
+      if (target[key] === undefined) {
+        // Guard against prototype pollution by checking if the key is a non-enumerable property
+        if (Object.prototype.hasOwnProperty.call(target, key)) {
```

Comment:
Keep in mind the prototype pollution check will cause keys to not be assigned unless the target object explicitly has  the keys, so if a new key is trying to be added this wouldn't work. I think that'll be problematic, looking at the hook `target` is passed in as an empty object, so no new keys will get added.

Reply:
So remove the checks?